Address physical constants for ccpp-physics PR#250

grantfirl opened the pull request:
@AndersJensen-NOAA, this PR is needed for https://github.com/ufs-community/ccpp-physics/pull/250 to address the physical constants coming through the host for the CCPP. In order to set them from the host, some constants in module_mp_tempo_params need to have the `parameter` attribute removed, making them variables. For any derived constants that subsequently use the newly made variables, they have to be initialized in a subroutine since Fortran doesn't allow executable statements bare within modules -- this is why the new mp_tempo_params_init() subroutine exists in that module. This has the knock-on effect of needing to call this subroutine from wherever tempo_init() is called. I tried to do this in various drivers here and in the ccpp-physics.

In conjunction with https://github.com/ufs-community/ccpp-physics/pull/250, when called through the CCPP, the parameters in module_mp_tempo_params will continue to initially get their default values. During the mp_tempo_init() call, tempo_init() is called, which calls mp_tempo_params_init(), calculating the initial values of derived constants. Then, after the tempo_init() call, whatever constants are available from the host are transferred to the module-level variables in module_mp_tempo_params. Finally, derived constants are re-derived using the values from the host.

Of course, outside of the CCPP, where you're not worrying about constant consistency, everything should continue to use values from module_mp_tempo_params and the derived constants are calculated during the tempo_init/mp_tempo_params_init() call.

You may want to double-check that this works in other instances where TEMPO is called (i.e. standalone, MPAS, etc.) before merging.

Synopsis of changes:
- remove PARAMETER attribute for physical constants that may come through the host (CCPP)
- add mp_tempo_params_init() subroutine to set derived constants
- call mp_tempo_params_init() from module_mp_tempo.F90 drivers

AndersJensen-NOAA commented:
@grantfirl Thank you for this! Can you merge the latest main into your branch? There two PR that should go in. #13 where I added freezing rain and #15 where I added a working test.  

There will be a merge convict -- you'll have to add these lines back to the standalone driver:
! Initialize physical constants
call mp_tempo_params_init()

Then your test should pass, and after that I will do a quick test using standalone MPAS.

grantfirl commented:
> @grantfirl Thank you for this! Can you merge the latest main into your branch? There two PR that should go in. #13 where I added freezing rain and #15 where I added a working test.
> 
> There will be a merge convict -- you'll have to add these lines back to the standalone driver: ! Initialize physical constants call mp_tempo_params_init()
> 
> Then your test should pass, and after that I will do a quick test using standalone MPAS.
> 
> Also, please update the version to 1.1.0 in the readme.

@AndersJensen-NOAA Done.

AndersJensen-NOAA commented:
@grantfirl merged.
